core/src/main/java/com/linecorp/armeria/common/unsafe/ByteBufHttpData.java

```diff
@@ -169,11 +181,22 @@ public String toString() {
 
     @Override
     public InputStream toInputStream() {
-        return new ByteBufInputStream(buf.retainedDuplicate(), true);
+        return new ByteBufInputStream(buf.duplicate(), false);
```

Comment:
Could you explain this change? `retainedDuplicate()` generates less garbage.

Reply:
I think we discussed this a long time ago, and ended up with it in https://github.com/line/armeria/pull/1938#issue-302381652

With `HttpData` closeable, it's simple for a user to reason about a reference needing to be contained in a single block and we're not working about `InputStream` moving out of the block, in other words, we don't want to create a new reference here that the user will have to close independenty of the `HttpData`.

I don't think there's any relation to garbage, it's just about whether a user can transfer the `InputStream` to a different scope and close it manually.

Reply:
Ah, OK. That's fine then. Thanks for elaboration. :bow: